modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -334,16 +335,23 @@ public MetadataMediator(RuntimeExtensionModel extensionModel, RuntimeComponentMo
       return success(subTypesUnion(output.getType(), subTypesMappingContainer));
     }
 
+    MetadataOutputResolver<?> outputResolver;
+    boolean isQuery = metadataKeyParts.stream().anyMatch(p -> p.getModelProperty(QueryModelProperty.class).isPresent());
+    if (isQuery && isDsqlQuery(key.getId())) {
+      outputResolver = new DsqlQueryMetadataResolver(componentModel);
+    } else {
+      outputResolver = resolverFactory.getOutputResolver();
```

Comment:
user a wrapper resolver that has this logic and is able to delegate to the dsql or native resolvers depending on the key when `getOutputMetadata` is invoked